src/Antiforgery/src/AntiforgeryTokenSet.cs

```diff
@@ -18,10 +18,10 @@ public class AntiforgeryTokenSet
         /// <param name="formFieldName">The name of the form field used for the request token.</param>
         /// <param name="headerName">The name of the header used for the request token.</param>
         public AntiforgeryTokenSet(
-            string requestToken,
-            string cookieToken,
+            string? requestToken,
+            string? cookieToken,
```

Comment:
Is this correct? I would expect these fields to be non-nullable. What's the use case for creating a token set with all null values. Also, why is the formfield non-nullable (apart from the check we do in the ctor)

Reply:
I've been trying to walk backwards from the implementation. For instance, in this case, `formFieldName` was the only value that disallowed nullability.

Reply:
I think we shouldn't be doing this mechanically and instead do what makes sense. I worry that if we just apply things mechanically we'll end up codifying behaviors that don't match the actual system expectations.

For example, it is very common that we avoid nullchecks in many cases but factually we still expect those types to not receive null values.

Reply:
I looked at this again, ad we anticipate these values to be null and have specific code paths for this: https://github.com/dotnet/aspnetcore/blob/master/src/Antiforgery/src/Internal/DefaultAntiforgery.cs#L115-L125